alerts/geomodel/alert.py

```diff
@@ -0,0 +1,135 @@
+from datetime import datetime
+import math
+from operator import attrgetter
+from typing import List, NamedTuple, Optional
+
+import netaddr
+
+from alerts.geomodel.config import Whitelist
+from alerts.geomodel.locality import State, Locality
+
+
+_AIR_TRAVEL_SPEED = 1000.0  # km/h
+
+_EARTH_RADIUS = 6373.0  # km # approximate
+
+_DEFAULT_SUMMARY = 'Authenticated action taken by a user outside of any of '\
+    'their known localities.'
+
+# TODO: Switch to dataclasses when we move to Python3.7+
+
+
+class Origin(NamedTuple):
+    '''A description of a location.
+    '''
+
+    city: str
+    country: str
+    latitude: float
+    longitude: float
+    geopoint: str
+
+
+class Alert(NamedTuple):
+    '''A container for the data the alerts output by GeoModel contain.
+    '''
+
+    source: str
+    category: str
+    type_: str
+    username: str
+    sourceipaddress: str
+    timestamp: datetime
+    origin: Origin
+    tags: List[str]
+    summary: str
+
+    def new(
+        username: str,
+        sourceip: str,
+        origin: Origin,
+        summary: str = _DEFAULT_SUMMARY
+    ) -> 'Alert':
+        '''Produce a new `Alert` with default values filled.
+        '''
+
+        return Alert(
+            source='geomodel',
```

Comment:
Many of these fields aren't used anywhere, (ex: source, type_, timestamp), and since we only use it in geomodel_alert as parameters to createAlertDict, what if we specified `tags` and `category` in the geomodel_alert, and used a dict for the other keys?